src/smvlang/smv_typecheck.cpp

```diff
@@ -1940,9 +1940,13 @@ void smv_typecheckt::create_var_symbols(
       else
         symbol.pretty_name = strip_smv_prefix(symbol.name);
 
-      symbol.value = nil_exprt{};
-      symbol.is_input = true;
+      if(symbol.type.id() == "submodule")
+        symbol.is_input = false;
+      else
+        symbol.is_input = true;
+
       symbol.is_state_var = false;
+      symbol.value = nil_exprt{};
```

Comment:
This line and the one above aren't actually necessary, the `symbolt` constructor will have set these already. (Though lines 1973 and 1975 should then also be removed.)